### PR TITLE
Fix 32558 - Moving Chef parser and importer to Core

### PR DESCRIPTION
**Pull request by domitea**
Like other parsers (https://github.com/theforeman/foreman/pull/8491, https://github.com/theforeman/foreman/pull/8508, https://github.com/theforeman/foreman/pull/8511), the Chef parser with importer is moving too. This PR adds them to core.

This PR relies on the Ansible one (https://github.com/theforeman/foreman/pull/8491) due to usage of initializer created in Ansible's PR.

### CODE REVIEW

**Comment by lzap** [comment]
Please add some integration tests - those which have a fixture file (JSON or YAML or whatever) and successfuly import a host.

File: test/unit/foreman_ansible/fact_parser_test.rb [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+      sample_mock.expects(:first_or_create)
+    end
+  end
+
+  # Tests for Network parser
+  class NetworkFactParserTest < ActiveSupport::TestCase
+    setup do
+      @facts_parser = AnsibleFactParser.new(
+        HashWithIndifferentAccess.new(
+          '_type' => 'ansible',
```
Comment (lzap) [inline test/unit/foreman_ansible/fact_parser_test.rb]: Ansible? is this a left over? I suggest to move these fixtures into separate files, we will be adding more over time.

**Comment by domitea** [comment]
I understand your concern and I know that testing is very important part of developing. I'm planning also refactor tests for parsers and importers together with refactoring of these components. That means I need firstly have all parsers and importers in one place for refactoring.

File: test/unit/foreman_ansible/fact_parser_test.rb (hunk 1 above)
Comment (domitea) [reply test/unit/foreman_ansible/fact_parser_test.rb]: Well, PR is based on commit of adding Ansible parser due to using own initializer defined in Ansible PR.

**Comment by lzap** [comment]
We are merging Ansible PR soon, unless there are any objections I am going to merge these parser PRs into core so @domitea  can start refactoring them.

**Review by ekohl — CHANGES_REQUESTED** [review]
Needs a rebase now.

File: config/environments/development.rb [hunk 2]
```diff
@@ -46,7 +46,7 @@
 
   if defined?(Bullet)
     config.after_initialize do
-      Bullet.enable = true
+      Bullet.enable = false
```
Comment (ekohl) [inline config/environments/development.rb]: Why is this needed?

**Review by ekohl — COMMENTED** [review]
Is there a license on the Chef image? Can we safely include it in our project?

File: config/environments/development.rb (hunk 2 above)
Comment (domitea) [reply config/environments/development.rb]: Nope, just a leftover from demoing...

**Comment by domitea** [comment]
I've updated PR. Conflict is resolved. 

@ekohl has a great comment about icon image of Chef. I cannot find this type of icon at any github repo. So I think that icon is under EULA of Chef. 
On the other hand, I found [this](https://github.com/chef/chef-web-docs/tree/master/themes/docs-new/static/images) folder at one of Chef's repo for documentation and that repo has Creative Commons licence. But I'm not a legal person so I'm not sure.

**Comment by ekohl** [comment]
Perhaps we should use https://www.chef.io/contact-us to ask them?

**Comment by tbrisker** [comment]
IANAL, but looks like the Chef logo falls under the trademark policy of their new owners Progress: https://www.progress.com/legal/trademarks which requires express written consent. I would suggest for now merging this using orange-C icon instead of the logo and asking for permission to use it.

**Comment by ares** [comment]
I asked this when I added it to foreman_chef plugin and got an approval. It's part of the same project, just a different repo now. Note that we already ship this in Foreman repos as part of foreman_chef package. So I don't think we shoud re-check.

**Comment by ekohl** [comment]
> I asked this when I added it to foreman_chef plugin and got an approval. It's part of the same project, just a different repo now.

Between then and now Chef was acquired by a different company. I'd like to be safe. Do you have any record of the approval?

> Note that we already ship this in Foreman repos as part of foreman_chef package. So I don't think we shoud re-check.

My concern is that previously we didn't ship it in commercial products (Red Hat Satellite, ATIX Orcharhino) while foreman.git is. That said, like @tbrisker IANAL.

**Comment by domitea** [comment]
Replacing the icon with the dummy one until licencing icons will be resolved. It can take a while so I think that PR is now ready for merge. When the we got approve for use of icon, I'll do it in another PR.

**Review by ekohl — DISMISSED** [review]

File: app/models/fact_names/foreman_chef/fact_name.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+module ForemanChef
+  class FactName < ::FactName
+    def origin
+      'Chef'
+    end
+
+    def icon_path
+      'icons64x64/stub/goldenrod-c.png'
```
Comment (tbrisker) [inline app/models/fact_names/foreman_chef/fact_name.rb]: This should be 
```suggestion
      'icons16x16/stub/goldenrod-c.png'
```

Comment (domitea) [reply app/models/fact_names/foreman_chef/fact_name.rb]: Thanks! PR updated!

**Review by tbrisker — APPROVED** [review]
Thanks @domitea !

**Comment by lzap** [comment]
Thanks!
